### DISCUSSION TITLE
UDP encapsulation without NAT_DETECTION_*_IP notify

### DISCUSSION THREAD

**Discussion by T0lva** [strongswan/strongswan#977] [General]
Hello,
is there a way to configure strongswan to force UDP encapsulation of ESP and disable NATD_D_IP / NATD_D_IP notifies ?

Setting "forceencaps" token to "yes" in ipsec.conf does force UDP encapsulation, but it doesn't seem possible to both force UDP encapsulation and deactivate NAT detection notifies.

Thanks !
Thomas

**Comment by tobiasbrunner** [comment]
No.

**Reply by T0lva** [reply]
ok, thanks for your answer !

**Reply by tobiasbrunner** [reply]
strongSwan doesn't have an option to disable NAT-D/T anyway, and forcing UDP-encap without it seems tricky. How would the other peer know that it should be used (or did you mean to configure this on both sides)? If so what's the problem with `forceencaps` and NAT-D?

**Reply by T0lva** [reply]
Yes, i'm in a situation where both peers are configured with `forceencaps=yes`.
If i understand well nat-detection mechanism as it is described in RFC 7296, this mechanism is useless in this situation, no ?

**Reply by tobiasbrunner** [reply]
Why?
